sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryStatisticsPage.scala

```diff
@@ -199,49 +202,98 @@ private[ui] class StreamingQueryStatisticsPage(parent: StreamingQueryTab)
+          <td class={"aggregated-num-total-state-rows-timeline"}>{graphUIDataForNumberTotalRows.generateTimelineHtml(jsCollector)}</td>
+          <td class={"aggregated-num-total-state-rows-histogram"}>{graphUIDataForNumberTotalRows.generateHistogramHtml(jsCollector)}</td>
         </tr>
-      // scalastyle:on
+          <tr>
+            <td style="vertical-align: middle;">
+              <div style="width: 160px;">
+                <div><strong>Aggregated Number Of Updated State Rows {SparkUIUtils.tooltip("Aggregated number of updated state rows.", "right")}</strong></div>
+              </div>
+            </td>
+            <td class={"aggregated-num-updated-state-rows-timeline"}>{graphUIDataForNumberUpdatedRows.generateTimelineHtml(jsCollector)}</td>
+            <td class={"aggregated-num-updated-state-rows-histogram"}>{graphUIDataForNumberUpdatedRows.generateHistogramHtml(jsCollector)}</td>
+          </tr>
+          <tr>
+            <td style="vertical-align: middle;">
+              <div style="width: 160px;">
+                <div><strong>Aggregated State Memory Used In Bytes {SparkUIUtils.tooltip("Aggregated state memory used in bytes.", "right")}</strong></div>
+              </div>
+            </td>
+            <td class={"aggregated-state-memory-used-bytes-timeline"}>{graphUIDataForMemoryUsedBytes.generateTimelineHtml(jsCollector)}</td>
+            <td class={"aggregated-state-memory-used-bytes-histogram"}>{graphUIDataForMemoryUsedBytes.generateHistogramHtml(jsCollector)}</td>
+          </tr>
+          <tr>
+            <td style="vertical-align: middle;">
+              <div style="width: 160px;">
+                <div><strong>Aggregated Number Of State Rows Dropped By Watermark {SparkUIUtils.tooltip("Aggregated number of state rows dropped by watermark.", "right")}</strong></div>
+              </div>
+            </td>
+            <td class={"aggregated-num-state-rows-dropped-by-watermark-timeline"}>{graphUIDataForNumRowsDroppedByWatermark.generateTimelineHtml(jsCollector)}</td>
+            <td class={"aggregated-num-state-rows-dropped-by-watermark-histogram"}>{graphUIDataForNumRowsDroppedByWatermark.generateHistogramHtml(jsCollector)}</td>
+          </tr>
+        // scalastyle:on
+
+      result ++= generateAggregatedCustomMetrics(query, minBatchTime, maxBatchTime, jsCollector)
+      result
     } else {
       new NodeBuffer()
     }
   }
 
+  def generateAggregatedCustomMetrics(
+   query: StreamingQueryUIData,
+   minBatchTime: Long,
+   maxBatchTime: Long,
+   jsCollector: JsCollector): NodeBuffer = {
+    val result: NodeBuffer = new NodeBuffer
+
+    // This is made sure on caller side but put it here to be defensive
+    require(query.lastProgress.stateOperators.nonEmpty)
+    val disabledCustomMetrics = Utils.stringToSeq(SQLConf.get.disabledStreamingCustomMetrics)
```

Comment:
> I think it'd be better to do opposite, disable everything and let end users enable it, as in many cases default metrics would be sufficient to get the picture of current status. Custom metrics are more likely auxiliary.

It makes sense, switching.

> So the ideal approach would be providing metric name with aggregation method (sum, avg, min, max), but if it sounds to be complicated just for this functionality, we could simply restrict to sum and explain it to config doc.

I think it's a good idea to have such functions but the question is what exactly.
Sum and max makes sense but the other 2 I can't really imagine a use-case.
Maybe we can wait for real use-cases and add aggregation methods when it's clear.

Reply:
If there's a custom metric about "latency", avg is probably the better aggregation instead of sum, though in such case we actually want the histogram with various percentiles. Let's restrict the aggregation to only "sum" with explanation on the config.

Reply:
Makes sense, we can come back to this when user need is more clear.

Reply:
I'm leaving this open because something is not 100% with the SQL conf reading from the UI side and test fails when I try to use `withSQLConf`.

Reply:
Hmm, now I see what's the matter. SparkUI doesn't have active session which is fine because it's common but w/o active session `SQLConf` parameters not propagated.

Reply:
I've moved the config to the UI section because of my previous comment.